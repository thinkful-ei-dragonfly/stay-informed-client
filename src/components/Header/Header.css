header {
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #000000;
  color: #ffffff;
  font-family: 'Titillium Web', sans-serif;
}

header a {
  margin: 0 5px;
  text-decoration: none;
  color: #ffffff;
}

header a:hover {
  text-decoration: underline;
}

h1 {
  font-size: 25px;
}