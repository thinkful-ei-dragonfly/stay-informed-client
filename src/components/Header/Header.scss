@import '../../index.scss';

.appHeader {
  background-color: #fafaf8;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: sticky;
  position: -webkit-sticky;
  position: -moz-sticky;
  position: -ms-sticky;
  position: -o-sticky;
  padding: 20px;
  background-color: $background;
  top: 0;
  z-index: 99999;
  // border: 1px solid;
  margin: 0 -25px;
  border-bottom: 1px solid;

  @media (min-width: 768px) {
    border-bottom: 0;
    margin: 0 auto;
    padding: 20px 0;
  }
  a {
    color: inherit;
    text-decoration: none;
  }
  .Header-Logo {
    font-size: 1.25em;
    margin-bottom: 0.5em;
    a {
      background: transparent;
      cursor: pointer;
    }
  }

  nav {
    .login-link,
    .search-nav,
    .register-link {
      @media (min-width: 768px) {
        padding: 10px;
        transition: color .25s ease, background .25s ease;
        border-radius: 10px;
      }

    }
    .login-link {
      margin-left: 25px;
      &:hover {
        background: pink;
        color: $red;
      }
    }
    .search-nav {
      &:hover {
        background: $lightgreen;
        color: $green;
      }
    }
    .register-link {
      margin-left: 25px;
      &:hover {
        color: $blue;
        background: $lightblue;
      }
    }

  }

  .signed-in {
    display: flex;

    nav {
      @media (min-width: 768px) {
        margin-left: 25px;
      }
      .header-user-name,
      .search-nav,
      .logout {
        padding-right: 25px;
        @media (min-width: 768px) {
          padding: 10px;
        }
        transition: color .25s ease, background .25s ease;
        border-radius: 10px;
      }
      .header-user-name {

        &:hover {
          color: $blue;
          background: $lightblue;
        }
        @media (min-width: 768px) {
          margin-left: 25px;
        }
      }
      .search-nav {
        &:hover {
          background: $lightgreen;
          color: $green;
        }
      }
      .logout {
        @media (min-width: 768px) {
          margin-left: 25px;
        }
        &:hover {
          background: $pink;
          color: $red;
        }
      }
    }
  }

  @media (min-width: 768px) {
    flex-direction: row;
  }
}
