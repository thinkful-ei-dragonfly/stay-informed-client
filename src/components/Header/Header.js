import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import UserContext from '../../contexts/UserContext'
import Search from './search.svg'
import Settings from './settings.svg'
import './Header.scss'

class Header extends Component {
  static contextType = UserContext

  handleLogoutClick = () => {
    this.context.processLogout()
  }

  renderLogoutLink() {
    return (
      <div className='header-controls signed-in'>
        <nav role="navigation">
          <Link
            className='header-user-name'
            to={`/user/${this.context.user.id}`}>
            <span className="full">{this.context.user.name}</span>
            <span className="mobile"><img src={Settings} alt='Settings'/></span>
          </Link>
          <Link
            className="search-nav"
            to='/search'>
            <span className="full">Quick Search</span>
            <span className="mobile"><img src={Search} alt='Search'/></span>
          </Link>
          <Link
            className='logout'
            onClick={this.handleLogoutClick}
            to='/login'>
            Logout
          </Link>
        </nav>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='header-controls signed-out' >
        <nav role="navigation">
          <Link
            className="search-nav"
            to='/search'>
            <span className="full">Quick Search</span>
            <span className="mobile"><img src={Search} alt='Search'/></span>
          </Link>
          <Link className='login-link'
            to='/login'>
            Login
          </Link>
          {''}
          <Link className='register-link'
            to='/register'>
            <span className="full">Sign up</span>
            <span className="mobile">Sign up</span>
          </Link>
        </nav>
      </div>
    )
  }

  render() {
    return (
      <header className='appHeader' role="banner">
        <h1 className='Header-Logo'>
          <Link to='/'>
            <span className='logo-red'>Stay</span><span className='logo-blue'>Informed</span>
          </Link>
        </h1>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </header>
    );
  }
}

export default Header
