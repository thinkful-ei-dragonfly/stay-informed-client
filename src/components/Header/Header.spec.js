import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './Header'

describe(`Header component`, () => {
  it('renders a form.Header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
