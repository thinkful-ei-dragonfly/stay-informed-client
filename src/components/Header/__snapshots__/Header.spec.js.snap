// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a form.Header by default 1`] = `
<header
  className="App-Header"
>
  <h1
    className="Header-Logo"
  >
    <Link
      to="/"
    >
      <span
        className="logo-red"
      >
        Stay
      </span>
      <span
        className="logo-blue"
      >
        Informed
      </span>
    </Link>
  </h1>
  <div
    className="header-controls signed-out"
  >
    <nav>
      <Link
        className="login-link"
        to="/login"
      >
        Login
      </Link>
      <Link
        className="register-link"
        to="/register"
      >
        Sign up
      </Link>
    </nav>
  </div>
</header>
`;
